geomet/esri.py

```diff
@@ -0,0 +1,233 @@
+#  Copyright 2013 Lars Butler & individual contributors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import json
+import geomet
+
+def load(source_file):
+    """
+    Converts Esri Json File to GeoJSON.
+
+    :param source_file:
+        Path to a file that contains the Esri JSON data.
+    
+    :returns: 
+         A GeoJSON `dict` representing the geometry read from the file.
+
+    """
+    return json.load(source_file)
+
+def loads(string):
+    """
+    Construct a GeoJSON `dict` from Esri JSON (string/dict).
+
+    :param string:
+        The Esri JSON geometry representation
+
+    :returns: 
+         A GeoJSON `dict` representing the geometry read from the file.
+    """
+    if isinstance(string, str):
+        string = json.loads(string)
+    if 'rings' in string:
+        return _esri_to_geojson_convert['rings'](string)
+    elif 'paths' in string:
+        return _esri_to_geojson_convert['paths'](string)
+    elif 'x' in string or 'y' in string:
+        return _esri_to_geojson_convert['x'](string)
+    elif 'points' in string:
+        return _esri_to_geojson_convert['points'](string)
+    else:
+        raise geomet.InvalidGeoJSONException('Invalid EsriJSON: %s' % string)
+
+        
+
+def dump(obj, dest_file, srid=None):
+    """
+    Converts GeoJSON to Esri JSON File.
+    """
+    return json.dump(dumps(obj, srid=srid), dest_file)
+
+def dumps(obj, srid=None):
+    """
+    Dump a GeoJSON-like `dict` to a Esri JSON.
+
+    :param string:
+        The GeoJSON geometry representation
+    :param int:
+        The default SRID value if none is present.
```

Comment:
I wonder if there's value in moving SRID checking upstream into this function, and making sure that the user doesn't pass in an SRID that's different than one which is present on the geometry. Maybe that's getting too involved, but if I was converting geometry and I thought it was 4326 and it contained metadata saying that it was 3857...I'd probably want to know.

Reply:
Added the test for the SRID when coming from GeoJSON.  The `EPSG:` needs to be replaced instead of `EPSG`.